scheds/rust/scx_rusty/src/bpf/main.bpf.c

```diff
@@ -425,8 +425,7 @@ static struct task_struct *tptr_to_task(u64 tptr)
 
 	if (err_task)
 		scx_bpf_error("Failed to retrieve task_struct for tptr %llu", tptr);
-	if (p)
-		return p;
+	return p ? p : NULL;
```

Comment:
Silly question, why are either of these needed? If `bpf_profile_read` fails it should hit the `scx_bpf_error` (which maybe should return null) and then it should be fine to return `p` always.

Reply:
Yeah you're right, I think `return p` directly is enough, failures should be catch by line 426.
I'll make an update and try to test the number of branches we can save from this.